lib/entities/webhook.ts

```diff
@@ -0,0 +1,215 @@
+import cloneDeep from 'lodash/cloneDeep'
+import { freezeSys, toPlainObject } from 'contentful-sdk-core'
+import enhanceWithMethods from '../enhance-with-methods'
+import errorHandler from '../error-handler'
+import { createUpdateEntity, createDeleteEntity } from '../instance-actions'
+import { AxiosInstance } from 'axios'
+import { CollectionProp, DefaultElements, MetaSysProps } from '../common-types'
+
+const entityPath = 'webhook_definitions'
+
+export type WebhookProps = {
+  /**
+   * System metadata
+   */
+  sys: MetaSysProps
+
+  /**
+   * Webhook name
+   */
+  name: string
+
+  /**
+   * Webhook url
+   */
+  url: string
+
+  /**
+   * Username for basic http auth
+   */
+  httpBasicUsername: string
+
+  /**
+   * Password for basic http auth
+   */
+  httpBasicPassword: string
+
+  /**
+   * Headers that should be appended to the webhook request
+   */
+  headers: {
+    [key: string]: string
+  }
+
+  /**
+   * Topics the webhook wants to subscribe to
+   */
+  topics: string[]
+
+  /**
+   * Transformation to apply
+   */
+  transformation?: {
+    method?: string
+    contentType?: string
+    includeContentLength?: boolean
+    body?: object
+  }
+}
+
+export interface WebHooks extends WebhookProps, DefaultElements<WebhookProps> {
```

Comment:
I think everywhere else we do "Webhooks" with a lowercase H? Could be wrong

Reply:
It's also the only place where an entity is in plural form 🤷‍♂️ Didn't want to break BC so I sticked with it.

Reply:
I also avoid doing breaking changes for type definitions. Some names are random but I'd collect all these things for the time being a fix them later.